src/client/preview/client_api.js

```diff
@@ -67,10 +99,10 @@ export default class ClientApi {
             return decorated(context);
           }, context);
         };
-      }, getStory);
+      }, storyFn);
 
       // Add the fully decorated getStory function.
-      this._storyStore.addStory(kind, storyName, fn);
+      this._storyStore.addStory(kindConfig, storyName, { storyFn: fn, storyMeta });
```

Comment:
Wouldn't it be better to have such interface here?

`addStory(kindName, storyName, fn, kindMeta, storyMeta);`

this is only my assumption though.